app/labor/markdown_parser.rb

```diff
@@ -138,9 +138,12 @@ def escape_colon_emojis_in_codeblock(html)
   end
 
   def catch_xss_attempts(markdown)
-    bad_xss = ['src="data', "src='data", "src='&", 'src="&', "data:text/html"]
-    bad_xss.each do |xss_attempt|
-      raise ArgumentError, "Invalid markdown detected" if markdown.include?(xss_attempt)
+    bad_xss_regex = [
+      Regexp.new("src=[\"'](data|&)", Regexp::IGNORECASE),
```

Comment:
It's a bit more common in Ruby to use regex literals with modifiers at the end, e.g. `/src=[\"'](data|&)/i`. You can also use a special form if you don't want to escape forward slashes in the regex: `%r{data:text/html[,;][\sa-z0-9]*}i`. 

I have no problem with the current version and think it reads well, but we don't use `Regexp.new` anywhere else in the codebase and there's something to be said for consistency. I'll leave the decision up to you.